File: two_pointers/0011_container_with_most_water.py
# https://leetcode.com/problems/container-with-most-water/

class Solution:
    def maxArea(self, height: list[int]) -> int:
        result = 0

        left, right = 0, len(height) - 1

        while left < right:
            area = (right - left) * min(height[left], height[right])
            result = max(result, area)

            if height[left] > height[right]:
                right -= 1
            else:
                left += 1

        return result

File: two_pointers/0011_container_with_most_water.cpp
// https://leetcode.com/problems/container-with-most-water/

#include <vector>

class Solution {
public:
    int maxArea(std::vector<int>& height) {
        int result = 0;

        int left = 0;
        int right = static_cast<int>(height.size()) - 1;

        while (left < right) {
            int area = (right - left) * std::min(height[left], height[right]);
            result = std::max(result, area);

            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }

        return result;
    }
};
